lib/dialogs/import/dropzone/index.tsx

```diff
@@ -27,33 +20,11 @@ function ImporterDropzone({
   const [tooManyFiles, setTooManyFiles] = useState();
 
   const handleAccept = (acceptedFiles) => {
-    const importers: Array<importer> = [
-      {
-        name: 'simplnote',
-        errorMessage:
-          'Only one Simplenote file (.json) can be imported at a time.',
-        fileTypes: ['json'],
-        count: 0,
-        multiple: false,
-      },
-      {
-        name: 'evernote',
-        errorMessage:
-          'Only one Evernote file (.enex) can be imported at a time.',
-        fileTypes: ['enex'],
-        count: 0,
-        multiple: false,
-      },
-      {
-        name: 'text',
-        errorMessage: '',
-        fileTypes: ['txt', 'md'],
-        count: 0,
-        multiple: true,
-      },
-    ];
-
     const filteredFiles = [];
+    const fileCounts = new Map();
+    importers.forEach((importer) => {
+      fileCounts.set(importer.name, 0);
+    });
```

Comment:
it took me a minute to understand the flow of data here. there are a few ways that I think it would be clearer.

_move the import finder into the new imports module_:

```ts
import * as importers from '../importers';

const seenImporters = new Set();
for (const file of acceptedFiles) {
	const importer = importers.forFilename(file);
	if (!importer) {
		setError('unrecognized_format');
		break;
	}

	if (!importer.multiple && seenImporters.has(importer)) {
		setError('too_many_files', importer.errorMessage);
		break;
	}

	seenImporters.add(importer);
	filteredFiles.push(file);
}
```

I believe that some of the complexity here is pointing to a mismatch with the model. If we step back a few levels, what is the goal of this function? What is wrong with importing multiple JSON files, or multiple ENEX files? Is the `multiple` complexity introduced because we had to always permit the upload dialog to allow multiple selections? It looks like if we can currently set multiple errors and keep on going; also that we can theoretically fail to find an importer for the dropped file and continue on from the `importers.find` statement.

Reply:
Thank you @dmsnell. The idea of using a Set here and moving the finder into the imports module makes total sense.

In the current iteration of the importers, there are three different import screens, one for each type, text files, Simplenote export, and Evernote export. The only one that allows multiple files to be uploaded is the text files. The Simplenote and Evernote importers are set to process only the first file of their type even if multiple are managed to be uploaded. I'm not sure what the reasoning for this was, but I'm guessing maybe performance concerns, especially around the Evernote exports. The interface would only allow you to choose one file though.

With this design change, there is only one import screen that should be able to accept any of the supported file types. So previously we depended on the dropzone to only allow one Simplenote or Evernote export file to be selected and uploaded at a time. I don't see a way to do that while also allowing multiple text files at the same time. That's why I chose this way of filtering the files and then only processing one of each and discarding the rest.

Reply:
maybe we should "see what happens" if we allow queueing up multiple imports for Simplenote and Evernote exports. it might work if we queue them up to run one-after-another.

then again, if that doesn't "just work" then it would probably be best to stop the process in its tracks and alert that we received multiple `.enex` files but can only use one yada yada

Reply:
I took a look today at allowing multiple SImplenote and Evernote files at once and it would require a refactor of the importers themselves to allow that. Maybe it is something we can look at in a separate PR.

For now I think the flow works pretty well. If they add more than one Simplenote or Evernote file it doesn't stop the flow and then have them go back and re-add the files they do want to import. Instead it shows them a list of the files which will be imported and gives a warning showing the files which won't be imported. 
This at least allows them to get some notes imported, then they could import separately the ones included in the warning.

My guess, but have not looked for data yet to back this up, is that people wouldn't normally have multiple Simplenote or Evernote backups that they are looking to import so likely not something that will impact anyone very often.

Reply:
I've been taking another look at allowing multiple json and enex files to be imported at the same time. The Simplenote importer seems to work fine as expected. But the Evernote one I'm struggling with.
I am doing simple tests with two enex files each with two notes in them. If I upload one at a time it works as expected. If I try uploading them both sometimes all the notes will duplicate and eight notes will be imported instead of the expected four.
I'm pretty sure it is something in the [desktop importer](https://github.com/Automattic/simplenote-electron/blob/develop/desktop/evernote-import/index.js) stream.
Do you have any theories or ideas @dmsnell?

Reply:
Thanks for looking at this with me today @dmsnell It was super helpful.